src/main/drivers/dshot_bitbang.c

```diff
@@ -77,8 +77,8 @@ dshotBitbangStatus_e bbStatus;
 #define BB_OUTPUT_BUFFER_ATTRIBUTE FAST_DATA_ZERO_INIT
 #define BB_INPUT_BUFFER_ATTRIBUTE  FAST_DATA_ZERO_INIT
 #elif defined(STM32H7)
-#define BB_OUTPUT_BUFFER_ATTRIBUTE DMA_RAM
-#define BB_INPUT_BUFFER_ATTRIBUTE  DMA_RAM
+#define BB_OUTPUT_BUFFER_ATTRIBUTE __attribute__((aligned(32)))
```

Comment:
This should still keep the `DMA_RAM` attribute - this is used on other MCU types as well. And not sure if the alignment is needed for H7 only (also, the linker actually aligns these already if `DMA_RAM` is used:
https://github.com/betaflight/betaflight/blob/ae62f4695837a2a6c5345862c4513aabf0f2a8db/src/link/stm32_flash_h750_1m.ld#L59

Reply:
The ALIGN(32) just aligns the start of the section. We need alignment of this buffer within the section. These definitions are only for H7. We don't need to move this out of normal RAM into D2_RAM, so only the aligned attribute is needed. Access to D2_RAM is 32bit, whereas for (D1)RAM its 64bit. DMAing data in/out of the ports isn't a concern re speed, however given that cache clean/invalidate operations are required, and the processor also needs to access the memory, it's better, performance wise, to keep it in D1.

Reply:
I should add that this should really be in DTCM in which case we don't need to worry about cache alignment as it's non-cached and very fast. Alas that's what kills the bootloader at the moment, so I'd like to revisit this once I've had a chance to figure that out and raise another PR.

Reply:
If i set this both back to  `DMA_RAM` and modified the `stm32_h750_common.ld` above it will fix #10322

Reply:
@TheIsotopes Unfortunately RAM overflows on an SPRACINGH7ZERO build as .bss overflows the 64K RAM. Ideally this should be in DTCM (which is 128K bytes), but the SP Racing bootloader then has issues. What you've tried gives us a clue though. Let me take a look at the possibility of cache coherency issues here.

Reply:
If I use the DMA_RAM attribute, this puts the buffers in the cacheable, but shareable region of memory. Shareable means "write though" so memory is always consistent with the cache. This slows the memory down by about a factor of 3 (on the H7), but ensures there should be no cache coherency issues. Unfortunately I find that the DSHOT telemetry is broken if I do this, and the speed control to the motors is a bit stuttery. Do you get RPM readings back if you enable the motors on the motors tab?

Reply:
@TheIsotopes Could you please give this branch another go. I've cache aligned the DSHOT buffers and cleaned/invalidated them as appropriate.

Reply:
@SteveCEvans If i only set this back to DMA_RAM the esc's are initialized and all Motors spinning correct, but the telemetry does not work reliably.

![telme](https://user-images.githubusercontent.com/29702171/97960170-2f9a7280-1db1-11eb-8325-e7d812359d62.jpg)

Reply:
> Could you please give this branch another go. I've cache aligned the DSHOT buffers and cleaned/invalidated them as appropriate.

@SteveCEvans give me a hint to test this. what do I have to change and how?

Reply:
@SteveCEvans your latest (https://github.com/SteveCEvans/betaflight/commit/4b77eb8dafd487ff54a295a31492653e7d75d313) is not working for me.
Motors spinning a moment and then stoped, sliders back to zero and the esc is reinitialized.

Reply:
@TheIsotopes Thanks for testing. So the ESCs initialise, but then things go awry? I've asked @Hydra for an H7NANO to investigate this.